### PR TITLE
reimplement member list API which is needed for Episerver

### PR DESCRIPTION
**Pull request by cnorthwood**
This API is used by the rarely used "Manage member sites" function within Episerver, which was not identified in the initial build, but has now been identified as broken functionality.

### CODE REVIEW

File: swagger/v0/swagger.yaml [hunk 1]
```diff
@@ -1112,40 +1184,14 @@ paths:
...
                           additionalProperties: false
-                        membershipNumber:
-                          type: string
-                        name:
-                          type: string
-                        serialNumber:
-                          type: string
-                        email:
-                          type: string
-                        id:
```
Comment (suzannehamilton) [inline swagger/v0/swagger.yaml]: It looks like some of the properies have been removed from the `/api/v0/json/vacancy/list` endpoint. Is that intentional?

File: spec/swagger_helper.rb [hunk 2]
```diff
@@ -6,20 +6,20 @@
   # Specify a root folder where Swagger JSON files are generated
   # NOTE: If you're using the rswag-api to serve API descriptions, you'll need
   # to ensure that it's configured to serve Swagger from the same folder
-  config.swagger_root = Rails.root.join("swagger").to_s
+  config.openapi_root = Rails.root.join("swagger").to_s
```
Comment (suzannehamilton) [inline spec/swagger_helper.rb:9]: Just to check: is this about these deprecation warnings? https://github.com/rswag/rswag/pull/702

File: spec/requests/v0/vacancy_spec.rb [hunk 3]
```diff
@@ -97,7 +97,7 @@
         schema type: :object,
                properties: {
                  type: { type: :string, enum: %w[vacancies] },
-                 list: { type: :array, items: BureauDetailsSchema::VACANCY_LIST_SCHEMA }
+                 list: { type: :array, items: BureauDetailsSchema::MEMBER_LIST_SCHEMA }
```
Comment (suzannehamilton) [inline spec/requests/v0/vacancy_spec.rb]: Is this change intentional?

**Review by suzannehamilton — COMMENTED** [review]
The new endpoint looks good 👍 

Just a couple of things I wasn't sure I understood.

File: spec/swagger_helper.rb (hunk 2 above)
Comment (cnorthwood) [reply spec/swagger_helper.rb:9]: ah yes, I probably should have put those in a different commit but I ended up fixing them as I noticed them

File: swagger/v0/swagger.yaml (hunk 1 above)
Comment (cnorthwood) [reply swagger/v0/swagger.yaml]: nope, this was a mistake

File: spec/requests/v0/vacancy_spec.rb (hunk 3 above)
Comment (cnorthwood) [reply spec/requests/v0/vacancy_spec.rb]: no, this was a mistake

**Review by suzannehamilton — APPROVED** [review]
Looks good to me 🚀
